swingset/src/main/java/com/nqadmin/swingset/SSDBComboBox.java

```diff
@@ -1542,51 +1537,66 @@ public void setSelectedStringValue(final String _value) {
 	}
 
 	/**
-	 * Sets the value stored in the component.
+	 * Sets the selected ComboBox item according to the specified mapping/key.
 	 * <p>
 	 * If called from updateSSComponent() from a RowSet change then the Component
 	 * listener should already be turned off. Otherwise we want it on so the
 	 * ultimate call to setSelectedIndex() will trigger an update the to RowSet.
 	 * <p>
-	 * The mappings ArrayList will be null until execute is called so
-	 * <p>
 	 * Currently not a bean property since there is no associated variable.
 	 *
-	 * @param _value database record primary key value to assign to combobox
+	 * @param _value database record mapping/key
 	 */
-	public void setSelectedValue(final long _value) {
+	public void setSelectedMapping(final long _value) {
 
-		// TODO consider firing a property change
-		// TODO what happens if user tries to pass null or if getSelectedValue() is null?
+		// HANDLED BY UNDERLYING INFRASTRUCTURE: TODO consider firing a property change
+		// HANDLED: TODO what if user pass null or if getSelectedValue() is null?
 
 		// 2020-08-03: Removing conditional as this could be called when consecutive records
 		// have the same value and we want to make sure to update the editor Strings
 
 		// IF MAPPINGS ARE SPECIFIED THEN LOCATE THE SEQUENTIAL INDEX AT WHICH THE
 		// SPECIFIED CODE IS STORED
 
-		try (Model.Remodel remodel = comboInfo.getRemodel()) {
+		try (Model.Remodel remodel = optionModel.getRemodel()) {
+			// TODO: if getAllowNull, might be empty, but still have 1 item.
 			if (!remodel.isEmpty()) {
 				//final int index = mappings.indexOf(_value);
 				final int index = remodel.getMappings().indexOf(_value);
-				
-				if (index == -1) {
-					logger.warn(getColumnForLog() + ": Could not find a corresponding item in combobox for value of " + _value + ". Setting index to -1 (blank).");
+
+				SSListItem item;
+				if (index != -1) {
+					item = remodel.get(index);
+				} else {
+					item = nullItem;
+					if (item == null) {
+						logger.warn(getColumnForLog() + ": Could not find a corresponding item in combobox for value of " + _value + ". Setting index to -1 (blank).");
+					}
 				}
 				
+				setSelectedItem(item);
+				//setSelectedIndex(index);
+				
 				logger.trace("{}: eventList - [{}].", () -> getColumnForLog(), () ->  remodel.getItemList().toString());
 				logger.trace("{}: options - [{}].", () -> getColumnForLog(), () ->  remodel.getOptions().toString());
 				logger.trace("{}: mappings - [{}].", () -> getColumnForLog(), () ->  remodel.getMappings().toString());
-				
-				setSelectedItem(index == -1 ? null : remodel.get(index));
-				//setSelectedIndex(index);
 			} else {
 				logger.warn(getColumnForLog() + ": No mappings available for current component. No value set by setSelectedValue().");
```

Comment:
Can/should we call setSelectedItem(null) here?

Reply:
Do you mean in the case when the list is empty? I think null is selected automatically in that case by JComboBox. Wouldn't hurt to select null.

Reply:
Yes. When remodel.isEmpty()==true.